### PR TITLE
added documentation with sample for std/hash mod, added mod.ts

### PR DESCRIPTION
**Pull request by deepakshrma**
Added basic document for hash module with sample example
Added mod.ts to easy access to all hash classes.

### CODE REVIEW

File: std/hash/mod.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export { Md5 } from "./md5.ts";
+export { Sha1 } from "./sha1.ts";
+export * from "./sha3.ts";
+export { Sha256, HmacSha256 } from "./sha256.ts";
+export { Sha512, HmacSha512 } from "./sha512.ts";
```
Comment (ry) [inline std/hash/mod.ts:5]: Don't add this module

Comment (deepakshrma) [reply std/hash/mod.ts:5]: Understand! Thanks.

File: std/hash/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import { Sha256, HmacSha256 } from "https://deno.land/std/hash/sha256.ts";
+
+const sha256 = new Sha256().update("中文");
+console.log(sha256.hex()); // 72726d8818f693066ceb69afa364218b692e62ea92b385782363780f47529c21
+console.log(sha256.toString()); // Same as above, 72726d8818f693066ceb69afa364218b692e62ea92b385782363780f47529c21
+
+const key = "Hi There";
+const hmac = new HmacSha256(key).update("中文");
+console.log(hmac.hex()); // 2f1ee7f7e89d3d53deadc1324dd0016bf8b859c620a910b3f6dba7b1a519d520
+console.log(hmac.toString()); // Same as above, 2f1ee7f7e89d3d53deadc1324dd0016bf8b859c620a910b3f6dba7b1a519d520
```
Comment (ry) [inline std/hash/README.md:53]: The comments showing output are not useful

Comment (deepakshrma) [reply std/hash/README.md:53]: Should I move as a statement before code? i thought it would be too verbose.

Comment (deepakshrma) [reply std/hash/README.md:53]: How about this?

## Sha256 and HmacSha256

**Uses:**

Creating `Sha256` hash is simple. You can use `Sha256` class instance and update the digest. Calling `hex` method will return the sha256 in `hex` value. You can also use `toString` method.

Note: For `HmacSha256`, you can pass secret `key` while creating instance of the object.

Comment (ry) [reply std/hash/README.md:53]: just remove the comments - they make the lines longer than 80 characters - and it doesn't add to the example.

Comment (deepakshrma) [reply std/hash/README.md]: Done! Thanks.

**Pull request merged**